src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -138,7 +144,8 @@ private bool NamespaceElementFilter(QsNamespaceElement elem)
                 {
                     if (elem is QsNamespaceElement.QsCallable call)
                     {
-                        return BuiltIn.RewriteStepDependencies.Contains(call.Item.FullName) || this.SharedState.IntrinsicCallableSet.Contains(call.Item.FullName);
+                        return BuiltIn.RewriteStepDependencies.Contains(call.Item.FullName) ||
```

Comment:
Does this method return `true` when an element should be filtered out?

Reply:
This method returns `true` if an element needs to be retained, even if Monomorphization would otherwise remove it.

Reply:
Looking at the description of #757, it seems like some specializations are not included in the call graph in some instances when they should. Given this description, I haven't yet understood why the `KeepAllIntrinsics` is included as an AND condition rather than an OR, since the AND condition restricts even more the elements that are retained.

Reply:
The `KeepAllIntrinsics` option allows the `IntrinsicsCallableSet` to be retained here. If `KeepAllIntrinsics` is false, we no longer want to special-case the intrinsics to keep them around. This is the change migrated from #764. The issue from #757 is that when a specialization for a callable is entered into the graph, we want to make sure that all the specialization from that same callable make it into the graph as well. The changes in ConcreteCallGraphWalker.cs reflect this.

Reply:
Thanks for the explanation Scott!